### PR TITLE
Added splitByLength() between indices function to grelfunctions documentation

### LINKED ISSUE #3934: Update user manual and in-app documentation for splitByLength()
The documentation, specifically for splitByLength() does not inform the user that they have the possibility to only return values within a given range of indices, Iv since updated the document but wanted to make a bug report on it for submitting the PR.

ex: 
`"123456789012345678901234567890".splitByLengths(3,4,5)[0,2]`
`returns [ "123",  "4567" ]`

More about this issue can be seen in this thread: https://github.com/OpenRefine/OpenRefine/issues/3929

### PR DESCRIPTION
**Pull request by mgutierrez98**
Fixes #3934}

Changes proposed in this pull request:
Added splitByLength() between indices function to grelfunctions documentation 
- 
- 
-

### CODE REVIEW

**Comment by wetneb** [comment]
Thank you! You will first need to merge your changes with master.

**Comment by mgutierrez98** [comment]
@wetneb I should probably add to the original splitByLength document as opposed to a new line as such: 
splitByLengths(3,4,5)[0,2]
as for functionality, the program does currently split if indices are specified which is lacking in the documentation. 
I will however attempt to toss in what you stated as well for functionality when the source string is shorter than the slice parameters

File: docs/docs/manual/grelfunctions.md [hunk 1]
```diff
@@ -174,9 +174,9 @@ Returns a string converted to a number. Will attempt to convert other formats in
 
 Returns the array of strings obtained by splitting s by sep. The separator can be either a string or a regex pattern. For example, `"fire, water, earth, air".split(",")` returns an array of 4 strings: [ "fire", " water", " earth", " air" ]. Note that the space characters are retained but the separator is removed. If you include “true” for the preserveTokens boolean, empty segments are preserved.
 
-###### splitByLengths(s, n1, n2, ...)
+###### splitByLengths(s, n1, n2, ...)[i1,i2] 
 
-Returns the array of strings obtained by splitting s into substrings with the given lengths. For example, `"internationalization".splitByLengths(5, 6, 3)` returns an array of 3 strings: [ "inter", "nation", "ali" ]. Excess characters are discarded.
+Returns the array of strings, between i1 and (i2 - 1), obtained by splitting s into substrings with the given lengths. For example, `"internationalization".splitByLengths(5, 6, 3)[0,2]` returns an array of 2 strings: [ "inter", "nation" ]. "ali" is desgraded as it falls outside of the indices. Excess characters are also disregarded.
```
Comment (ostephens) [inline docs/docs/manual/grelfunctions.md:179]: I think the original misunderstanding reported in #3929 was that the user was expecting something like:

"University of Washington".splitByLengths(4)

to return
["Univ","ersity of Washington"]

but it actually returns
["Univ"]

I think what is needed here is just to make this even more explicit - so expanding on the existing comment "Excess characters are discarded" to make it extra clear - so taking what you have already written something like:

"The remaining characters (in this case "zation") are discarded and not included in the array output"

**Review by ostephens — CHANGES_REQUESTED** [review]
@thadguidry  had proposed improving the documentation that appears in the OpenRefine in-app "Help" - which is text that comes from the code rather than the documentation pages. You'll need to look in https://github.com/OpenRefine/OpenRefine/blob/master/main/src/com/google/refine/expr/functions/strings/SplitByLengths.java

It currently reads: "Returns the array of strings obtained by splitting s into substrings with the given lengths"

**Pull request closed without merging**